docs/standard/serialization/system-text-json/source-generation.md

```diff
@@ -93,14 +95,14 @@ Here are the preceding examples in a complete program:
 
 :::code language="csharp" source="snippets/source-generation/csharp/BothModesNoOptions.cs" id="All":::
 
-## Specify source generation mode
+## Specify source-generation mode
 
-You can specify metadata collection mode or serialization optimization mode for an entire context, which may include multiple types. Or you can specify the mode for an individual type. If you do both, the mode specification for a type wins.
+You can specify metadata-collection mode or serialization-optimization mode for an entire context, which may include multiple types. Or you can specify the mode for an individual type. If you do both, the mode specification for a type wins.
```

Comment:
I'm not sure if "metadata-collection" is the correct terminology. The source generator _collects_ type metadata at compile time in all cases, but then generates code that either populates contract metadata using the contract customization APIs (JsonTypeInfo and friends) and/or serializes using fully inlined methods.
```suggestion
You can specify metadata-based mode or serialization-optimization mode for an entire context, which may include multiple types. Or you can specify the mode for an individual type. If you do both, the mode specification for a type wins.
```